src/config/src/meta/stream.rs

```diff
@@ -554,6 +562,44 @@ impl PartialEq for DistinctField {
 }
 impl Eq for DistinctField {}
 
+#[derive(Debug, Clone, Default, Serialize, Deserialize, ToSchema, PartialEq)]
+pub struct TimeRange {
+    pub start: i64,
+    pub end: i64,
+}
+
+impl TimeRange {
+    pub fn new(start: i64, end: i64) -> Self {
+        Self { start, end }
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.start == 0 && self.end == 0
+    }
+
+    pub fn contains(&self, other: &Self) -> bool {
+        self.start <= other.start && self.end >= other.end
+    }
+
+    pub fn intersects(&self, other: &Self) -> bool {
+        self.start <= other.end && self.end >= other.start
+    }
+
+    pub fn split(&self, other: &Self) -> Vec<Self> {
+        if !self.intersects(other) {
+            return vec![self.clone()];
+        }
+
+        let mut ranges = Vec::new();
+        if self.start < other.start {
+            ranges.push(Self::new(self.start, other.start));
+        }
+        if self.end > other.end {
+            ranges.push(Self::new(other.end, self.end));
+        }
+        ranges
+    }
```

Comment:
Here, one issue, the delete range should include the end red date or not? for example, the red date is:
```
[2024-12-12, 2024-12-15]
```
and the original date is:
```
[2024-12-13, 2024-12-16]
```
then, we will generate delete range
```
[2024-12-15, 2024-12-16]
```
We will delete the data of `2024-12-15`, is this expect?

Reply:
not expected. I think i will pass in the red days range as [a, b+1] to this function and that should fix it.